From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Mon, 26 Jul 2021 19:35:00 +0200
Applied-Upstream: https://github.com/logrotate/logrotate/commit/a98c38bc867ec59e00625b48262bb3334c8f5728
Subject: Do not warn on key value pair separated by only an equal sign

Do not warn if a configuration directive is specified with the key and
value separated by just an equal sign, like:

    size=+2048k

The warning is intended for the usage of:

    size2048k

Fixes: 2b588b5e ("Log if keyword is not properly separated")
Fixes: #410
---
 config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.c b/config.c
index ca32cd6..ee10b4f 100644
--- a/config.c
+++ b/config.c
@@ -1097,7 +1097,7 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                                 configFile, lineNum);
                         continue;
                     }
-                    if (!isspace((unsigned char)*start)) {
+                    if (!isspace((unsigned char)*start) && *start != '=') {
                         message(MESS_NORMAL, "%s:%d keyword '%s' not properly"
                                 " separated, found %#x\n",
                                 configFile, lineNum, key, *start);
